docs/quickstart/network-runner.md

```diff
@@ -12,38 +18,525 @@ That repository's README details the tool.
 
+There are also binary releases ready to use at [releases](https://github.com/ava-labs/avalanche-network-runner/releases) ready to use.
+
+To build and install the binary locally (requires `golang` to be installed. Check the [requirements](https://github.com/ava-labs/avalanchego#installation) for the minimum version):
+```bash
+cd ${HOME}/go/src/github.com/ava-labs/avalanche-network-runner
+go install -v ./cmd/avalanche-network-runner
+```
+
 Unless otherwise specified, file paths given below are relative to the root of this repository. 
 
 ## Usage
 
-The basic pattern for using the Avalanche Network Runner is:
+There are two main ways to use the network-runner:
+* Run ANR as a binary
+  This is the recommended approach for most use cases. Doesn't require golang installation and provides a RPC server with an HTTP API and a client library for easy interaction.
+* Import this repository into your go program
+  This allows for custom network scenarios and high flexibility, but requires more code to be written. 
+
+Running the binary, the user can send requests to the RPC server in order to start a network, create subnets, add nodes to the network, remove nodes from the network, restart nodes, etc.. You can make requests through the `avalanche-network-runner` command or by making API calls. Requests are "translated" into gRPC and sent to the server.
+
+Each node can then also be reached via [api](https://github.com/ava-labs/avalanche-network-runner/tree/main/api) endpoints which each node exposes.
+
+The following diagram is a simplified view of the high level architecture of the tool:
+![ANR architecture](/img/grpc-networkrunner.svg "Simplified ANR Architecture")
+
+## Examples 
+
+When running with the binary, ANR runs a server process as an RPC server which then waits for API calls and handles them.
+Therefore we run one shell with the RPC server, and another one for issuing calls.
+
+### Start the server
+
+```bash
+avalanche-network-runner server \
+--log-level debug \
+--port=":8080" \
+--grpc-gateway-port=":8081"
+```
+
+Note that the above command will run until you stop it with `CTRL + C`. Further commands will have to be run in a separate terminal.
+
+The RPC server listens to two ports:
+* `port`: the main gRPC port (see [gRPC](https://grpc.io/)
+* `grpc-gateway-port`: the gRPC gateway port (see [gRPC-gateway](https://grpc-ecosystem.github.io/grpc-gateway/) ), which allows for HTTP requests
+
+When using the binary to issue calls, the main port will be hit. In this mode, the binary executes compiled code to issue calls.
+Alternatively, plain HTTP can be used to issue calls, without the need to use the binary. In this mode, the `grpc-gateway-port` should be queried.
+
+Each of the examples below will show both modes, claritying its usage.
+
+### Run queries
+
+#### ping the server
+
+```bash
+curl -X POST -k http://localhost:8081/v1/ping -d ''
+
+# or
+avalanche-network-runner ping \
+--log-level debug \
+--endpoint="0.0.0.0:8080"
+```
```

Comment:
Please use separate ```bash for `curl` and `avalanche-network-runner` so that it is easy to copy+paste the command. This applies to all the commands in this doc.